scripts/preinstall.js

```diff
@@ -17,7 +19,35 @@ const configText = `module.exports = {
   localPlugins: [],
 }`;
 
+// simple utility to remove whitespace from string
+function trim(string) {
+  return string.replace(/^\s+|\s+$/g, '');
+}
+
 try {
+  // check minimum version of npm
+  let npmVersion = execSync('npm --version', {
+    encoding: 'utf8'
+  });
+  let nodeVersion = execSync('node --version', {
+    encoding: 'utf8'
+  });
+
+  npmVersion = trim(npmVersion);
+  nodeVersion = trim(nodeVersion);
+
+  const npmMin = process.env.npm_package_engines_npm;
+  const nodeMin = process.env.npm_package_engines_node;
+
+  if (
+    !semver.satisfies(npmVersion.version, npmMin) &&
```

Comment:
Maybe I'm reading this wrong but should it be an OR here? So it'll throw if either one is the wrong version?

Reply:
Good catch. Updated.